gulpfile.js

```diff
@@ -1,29 +1,92 @@
+const { spawn } = require('child_process');
+
+const del = require('del');
 const forever = require('forever-monitor');
 const gulp = require('gulp');
+const gulpBabel = require('gulp-babel');
 const eslint = require('gulp-eslint');
 const mocha = require('gulp-mocha');
+const sourcemaps = require('gulp-sourcemaps');
+const webpackStream = require('webpack-stream');
+
+const webpackConfig = require('./webpack.config.js');
+
+/**
+ * @summary Delete the dist/ directory
+ */
+const babelClean = () => del(['dist']);
+
+/**
+ * @summary Copy files that don't need to be compiled from src/ to dist/
+ */
+const babelCopy = () => gulp.src(['src/**', '!src/**/*.js', '!src/tests/integration/**'])
+  .pipe(gulp.dest('dist'));
+
+/**
+ * @summary Transpile JavaScript files and place them in dist/. Also, generate sourcemaps
+ */
+const babelCompile = () => gulp.src(['src/**/*.js'])
+  .pipe(sourcemaps.init())
+  .pipe(gulpBabel())
+  .pipe(sourcemaps.write('.'))
+  .pipe(gulp.dest('dist'));
 
 /**
  * @summary Use Eslint linting *.js file besides source files in node_modules
  */
-gulp.task('lint', () => gulp.src(['**/*.js', '!node_modules/**'])
+const lint = () => gulp.src(['src/**/*.js', '**.js'])
```

Comment:
Shouldn't `**.js` be `*.js` if you want to specify all `.js` files in the current directory? I think it'll work either way and I can't seem to find a case where the two would be different but I'm pretty sure `*.js` makes more sense here as `**` means to look through recursive directories